docs/reference/reana-yaml/index.md

```diff
@@ -2,9 +2,115 @@
 
 ## About
 
-The ``reana.yaml`` reference guide.
+REANA workflow specification file should answer 4 main questions:
+
+* What is your input data?
+* Which code analyses it?
+* What is your environment?
+* Which steps did you take?
+
+You have to structure your research data analysis repository into input `data`
+ and `parameters`, runtime `code`, computing `environments`, and computational
+`workflows`.
 
 ## YAML syntax
 
+REANA workflow specification uses a human-readable data-serialization language
+YAML ("YAML Ain't Markup Language"). It uses Python-style indentation and supports
+basic data types such as mappings, sequences, scalars.
+More information can be found in [the official documentation](https://yaml.org/spec/1.2/spec.html).
+
+## Write reana.yaml
+
+With your favourite text editor create file `reana.yaml`
+
+### Serial
+
+```yaml
+  version: 0.3.0
+  inputs:
+    files:
+      - code/helloworld.py
+      - data/names.txt
+    parameters:
+      helloworld: code/helloworld.py
+      inputfile: data/names.txt
+      outputfile: results/greetings.txt
+      sleeptime: 0
+  workflow:
+    type: serial
+    specification:
+      steps:
+        - environment: 'python:2.7-slim:tag'
+          commands:
+            - python "${helloworld}"
+                --inputfile "${inputfile}"
+                --outputfile "${outputfile}"
+                --sleeptime ${sleeptime}
+  outputs:
+    files:
+     - results/greetings.txt
+```
+
+#### Inputs & Outputs
+
+Analysis inputs consists of two types: `files` and `parameters`.
+
+Inputs of type `files` or `directories` will be automatically uploaded into
+REANA while doing `reana-client run` or `reana-client upload`. The same applies
+for section`outputs`, `reana-client download` command will automatically
+download all files specified there. Files are defined in a list structure type.
+
+Inputs of type `parameters` will be applied for the `commands` specified in the
+`workflow` section. It uses a key value data structure.
+
+#### Workflow
+
+Workflow which represents the steps that need to be run to reproduce an analysis.
+
+You have to specify the type of a workflow. Currently REANA supports 3 workflow
+definition languages: serial, cwl, yadage.
+
+The example above uses the `serial` workflow definition language - `type: serial`
+
+Specification part consists of a list of steps and steps consist a list of
```

Comment:
```suggestion
The `specification` part consists of a list of steps and `steps` consist a list of
```

I find this sentence a bit confusing. One might think, why is there even a `steps` part if the `specification` part already consists of a list of steps?